src/Features/Core/Portable/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -709,30 +714,31 @@ private static ImmutableArray<ISymbol> GetOverriddenSymbols(ISymbol memberSymbol
                     includeHiddenLocations: false);
             }
 
-            if (locations.Length == 1)
+            if (locations is [var location])
             {
-                var location = locations[0];
                 if (location.IsInMetadata)
                 {
-                    return DefinitionItem.CreateMetadataDefinition(
+                    Contract.ThrowIfNull(symbol.ContainingAssembly);
+
+                    var metadataLocation = DefinitionItemFactory.GetMetadataLocation(symbol.ContainingAssembly, solution, out var originatingProjectId);
+
+                    return DefinitionItem.Create(
                         tags: [],
                         displayParts: [],
-                        nameDisplayParts: [],
-                        solution,
-                        symbol);
+                        sourceSpans: [],
+                        classifiedSpans: [],
+                        metadataLocations: [metadataLocation],
+                        properties: ImmutableDictionary<string, string>.Empty.WithMetadataSymbolProperties(symbol, originatingProjectId));
```

Comment:
why not still have the CreateMetadataDefinition helper to deal with a lot of these default behaviors?

Reply:
`CreateMetadataDefinition` doesn't make sense. There is generally no "metadata definition". There is a definition that has source locations and/or metadata locations.

`ToSlimDefinitionItem` is creating very specialized item that no other feature needs.